models/Models.py

```diff
@@ -29,7 +30,14 @@ def model_function(model_name):
         
             }
     return model_maker_dict[model_name]    
-def make_model(model_name, **args):    
+def make_model(model_name, **args):
+    m_fn = model_function(model_name)
+    if args and hasattr(m_fn,'parameter_range'):
+        provided = set(args.keys())
+        accepted = set([a.name for a in m_fn.parameter_range])
+        if not provided.issubset( accepted ):
+            print ("provided arguments",sorted(provided),"do not match the accepted ones",sorted(accepted))
+            sys.exit(-1)
```

Comment:
Is there any way to handle this more gracefully? Like raise an exception and catch that from the calling code and shutdown properly. Or if not possible, consider calling `MPI_Abort` instead. `exit(-1)` may not cause the entire MPI application to terminate and while this may not be an issue on our clusters, it may be costly on systems where we have a limited allocation.

Reply:
this is just when one builds the model by hand with 

`python3 BuildModel.py mnist_torch kernel_size=5 dense=107`

for example. This should never be caught in mpi-opt